Assets/MixedRealityToolkit/_Core/Utilities/Physics/TwoHandRotateLogic.cs

```diff
@@ -100,31 +69,209 @@ private static Vector3 ProjectHandlebarGivenConstraint(RotationConstraintType co
+        public TwoHandRotateLogic(RotationConstraintType constrainRotation)
+        {
+            rotationConstraint = constrainRotation;
+        }
+
+        /// <summary>
+        /// Initializes twohand system with controller/hand source info: 
+        /// the Dictionary collection already filled with controller/hand info,
+        /// and the Transform of the GameObject to be manipulated.
+        /// </summary>
+        /// <param name="handsPressedMap">Dictionary listing inputSourceStates</param>
+        /// <param name="manipulationRoot">Transform of gameObject to be two hand manipulated</param>
+        public void Setup(Dictionary<uint, Vector3> handsPressedMap, Transform manipulationRoot)
+        {
+            currentRotationConstraint = rotationConstraint;
+            previousHandlebarRotation = GetHandlebarDirection(handsPressedMap, manipulationRoot);
+        }
+
+        /// <summary>
+        /// Updates internal states based on current Controller/hand states.
+        /// </summary>
+        /// <param name="handsPressedMap">Dictionary listing inputSourceStates</param>
+        /// <param name="manipulationRoot">Transform of gameObject to be two hand manipulated</param>
+        /// <param name="currentRotation">New rotation to be applied</param>
+        /// <returns>Quaternion describing rotation based on position of inputSources - Controllers/Hands</returns>
+        public Quaternion Update(Dictionary<uint, Vector3> handsPressedMap, Transform manipulationRoot, Quaternion currentRotation)
+        {
+            var handlebarDirection = GetHandlebarDirection(handsPressedMap, manipulationRoot);
+            var handlebarDirectionProjected = ProjectHandlebarGivenConstraint(currentRotationConstraint, handlebarDirection, manipulationRoot);
+            var prevHandlebarDirectionProjected = ProjectHandlebarGivenConstraint(currentRotationConstraint, previousHandlebarRotation, manipulationRoot);
+            previousHandlebarRotation = handlebarDirection;
+
+            var rotationDelta = Quaternion.FromToRotation(prevHandlebarDirectionProjected, handlebarDirectionProjected);
+
+            var angle = 0f;
+            var axis = Vector3.zero;
+            rotationDelta.ToAngleAxis(out angle, out axis);
+            angle *= RotationMultiplier;
+
+            if (currentRotationConstraint == RotationConstraintType.YAxisOnly)
+            {
+                // If we are rotating about Y axis, then make sure we rotate about global Y axis.
+                // Since the angle is obtained from a quaternion, we need to properly orient it (up or down) based
+                // on the original axis-angle representation. 
+                axis = Vector3.up * Vector3.Dot(axis, Vector3.up);
+            }
+            return Quaternion.AngleAxis(angle, axis) * currentRotation;
+        }
     }
+
+
+
+
+
+
+
+    ///// <summary>
```

Comment:
is this code ever going to be needed? if not, please remove as it can lead to customer confusion. if it _is_ going to be required, please add a TODO comment re: why it is currently commented

Reply:
👍